metadata-store-directory-common/src/test/java/org/apache/helix/msdcommon/mock/TestMockMetadataStoreDirectoryServer.java

```diff
@@ -0,0 +1,118 @@
+    List<Map<String, Object>> routingDataList =
+        (List<Map<String, Object>>) resultMap.get(MetadataStoreRoutingConstants.ROUTING_DATA);
+    Collection<String> allRealms = routingDataList.stream().map(mapEntry -> (String) mapEntry
+        .get(MetadataStoreRoutingConstants.SINGLE_METADATA_STORE_REALM))
+        .collect(Collectors.toSet());
+    Assert.assertEquals(allRealms, routingData.keySet());
+    Map<String, List<String>> retrievedRoutingData = routingDataList.stream().collect(Collectors
+        .toMap(mapEntry -> (String) mapEntry
+                .get(MetadataStoreRoutingConstants.SINGLE_METADATA_STORE_REALM),
+            mapEntry -> (List<String>) mapEntry.get(MetadataStoreRoutingConstants.SHARDING_KEYS)));
+    Assert.assertEquals(retrievedRoutingData, routingData);
+
+    // Send a GET request for all realms
+    getRequest = new HttpGet(endpoint + MockMetadataStoreDirectoryServer.REST_PREFIX + namespace
+        + MockMetadataStoreDirectoryServer.ZK_REALM_ENDPOINT);
+    getResponse = httpClient.execute(getRequest);
+    Map<String, Collection<String>> allRealmsMap = MockMetadataStoreDirectoryServer.OBJECT_MAPPER
+        .readValue(getResponse.getEntity().getContent(), Map.class);
+    Assert
+        .assertTrue(allRealmsMap.containsKey(MetadataStoreRoutingConstants.METADATA_STORE_REALMS));
+    allRealms = allRealmsMap.get(MetadataStoreRoutingConstants.METADATA_STORE_REALMS);
+    Assert.assertEquals(allRealms, routingData.keySet());
+
+    // Send a GET request for testZkRealm
+    String testZkRealm = "zk-0";
+    getRequest = new HttpGet(endpoint + MockMetadataStoreDirectoryServer.REST_PREFIX + namespace
+        + MockMetadataStoreDirectoryServer.ZK_REALM_ENDPOINT + "/" + testZkRealm);
+    getResponse = httpClient.execute(getRequest);
+    Map<String, Object> shardingKeysMap = MockMetadataStoreDirectoryServer.OBJECT_MAPPER
+        .readValue(getResponse.getEntity().getContent(), Map.class);
+    Assert.assertTrue(
+        shardingKeysMap.containsKey(MetadataStoreRoutingConstants.SINGLE_METADATA_STORE_REALM));
+    Assert.assertTrue(shardingKeysMap.containsKey(MetadataStoreRoutingConstants.SHARDING_KEYS));
+    String zkRealm =
+        (String) shardingKeysMap.get(MetadataStoreRoutingConstants.SINGLE_METADATA_STORE_REALM);
+    Collection<String> shardingKeyList =
+        (Collection) shardingKeysMap.get(MetadataStoreRoutingConstants.SHARDING_KEYS);
+    Assert.assertEquals(zkRealm, testZkRealm);
+    Assert.assertEquals(shardingKeyList, routingData.get(testZkRealm));
+
+    // Try sending a POST request (not supported)
+    HttpPost postRequest = new HttpPost(
+        endpoint + MockMetadataStoreDirectoryServer.REST_PREFIX + namespace
+            + MockMetadataStoreDirectoryServer.ZK_REALM_ENDPOINT + "/" + testZkRealm);
+    CloseableHttpResponse postResponse = httpClient.execute(postRequest);
+
+    // Shutdown
+    server.stopServer();
+    httpClient.close();
```

Comment:
Minor as it is a test: maybe we would use try...finally to close the server and client, in case exception or assertion fails and server is still running?

Reply:
Sure. updated.